packages/universal-cookie/src/utils.ts

```diff
@@ -3,14 +3,24 @@ import { Cookie, CookieGetOptions } from './types';
 
 export function hasDocumentCookie() {
   // JSDOM does not support changing cookies, disable it for tests
-  if (isJsDom()) {
+  if (isJsDom() || cookieDisabledBySettings()) {
     return false;
   }
 
   // Can we get/set cookies on document.cookie?
   return typeof document === 'object' && typeof document.cookie === 'string';
 }
 
+function cookieDisabledBySettings() {
+  if (typeof navigator !== 'object') {
+      return true;
+  }
+
+  // user can disable cookie for site
+  // in browser settings
+  return !navigator.cookieEnabled;
```

Comment:
Just to be in the safe side, I would do a type-safe check `navigator.cookieEnabled === false`. I know it's supported by all major browser but it would make the check more accurate.